cpp/tools/parquet/parquet_dump_footer.cc

```diff
@@ -0,0 +1,230 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
```

Comment:
We can except arrow fs is a streaming abstraction which will make the logic more complex. I avoided it because mmap is easier for this case. That said if we want to support object stores that's the easiest way there. Do we want to support object stores natively?

Reply:
> We can except arrow fs is a streaming abstraction which will make the logic more complex

`FileSystem::OpenInputFile` provides a `io::RandomAccessFile` interface, wouldn't this be ok? And LocalFileSystem can contains mmap here.

> That said if we want to support object stores that's the easiest way there. Do we want to support object stores natively?

If we cannot compile it, maybe we should denote this is some platform only in CMake? Otherwise we should including fs logic

Reply:
Done.